packages/ui/src/components/TextArea/index.tsx

```diff
@@ -194,12 +199,27 @@ export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
 
     useEffect(() => {
       const textArea = textAreaRef.current
-      if (textArea && rows === 'auto') {
+      const padding = theme.space['1.5']
+
+      if (textArea && rows === 'auto' && !maxRows) {
         textArea.style.height = 'auto'
         textArea.style.resize = 'none'
-        textArea.style.height = `${textArea.scrollHeight === 61 ? 46 : textArea.scrollHeight + 2}px`
+        textArea.style.height = `${textArea.scrollHeight + 2}px`
+      } else if (textArea && maxRows) {
+        const lineHeight = parseFloat(getComputedStyle(textArea).lineHeight)
+
+        textArea.style.height = 'auto'
+        const maxHeight = maxRows * lineHeight
+
+        textArea.style.height = `${textArea.scrollHeight + 2}px`
+        textArea.style.maxHeight = `calc(${maxHeight}px + ${padding} + ${padding}`
```

Comment:
```suggestion
        textArea.style.maxHeight = `calc(${maxHeight}px + ${2 * padding}`
```

Reply:
padding is a string

Reply:
```suggestion
        textArea.style.maxHeight = `calc(${maxHeight}px + 2* (${ padding})`
```

Should work this way then